### PR TITLE
Moves the crawler code to a separate module.

### PR DESCRIPTION
**Pull request by timuralp**
Allows us to avoid monkey patching eventlet when only the exceptions are imported.

### CODE REVIEW

File: test/unit/test_container_crawler.py [hunk 1]
```diff
@@ -7,15 +7,15 @@
 import time
 import unittest
 
-import container_crawler
-from container_crawler import RetryError
+import container_crawler.crawler as crawler
```
Comment (thiagodasilva) [inline test/unit/test_container_crawler.py]: could it be `from container_crawler import crawler` ?

Comment (timuralp) [reply test/unit/test_container_crawler.py]: Done

File: container_crawler/crawler.py [hunk 2]
```diff
@@ -0,0 +1,393 @@
...
+            self._outstanding -= 1
+            if error and not self._error:
+                self._error = error
+            if retry and not self._retry:
+                self._retry = retry
+            if self._outstanding == 0:
+                self._done.notify()
+
+
+class Crawler(object):
```
Comment (thiagodasilva) [inline container_crawler/crawler.py]: Now that I looked at the 1space PR that I noticed the changed name. IMHO, ContainerCrawler seems more descriptive. Crawler is a shorter name, but nowadays "crawler" is just too commonly used, so making it more indicative of what kind of crawler is helpful.

Comment (timuralp) [reply container_crawler/crawler.py]: On the other hand, it is in the container_crawler module, so it's always imported as `container_crawler.crawler.Crawler`. If we spell it out in all three places, we end up with `container_crawler.container_crawler.ContainerCrawler`. What do you think?

**Pull request merged**